File: 242-valid-anagram/valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False

        freq = {}

        for i in s:
            if i not in freq:
                freq[i] = 1
            else:
                freq[i] += 1

        for i in t:
            if i not in freq:
                return False
            else:
                freq[i] -= 1

        for value in freq.values():
            if value != 0:
                return False

        return True

File: 242-valid-anagram/valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if(s.length()!=t.length()){
            return false;
        }
        HashMap<Character, Integer> freq = new HashMap<>();

        for(char ch : s.toCharArray()){
            if(!freq.containsKey(ch)){
                freq.put(ch,1);
            }
            else{
                freq.put(ch, freq.get(ch) + 1);

            }
        }

        for(char ch : t.toCharArray()){
            if(!freq.containsKey(ch)){
                return false;
            }
            else{
                freq.put(ch, freq.get(ch) - 1);
            }
        }

        for (int value : freq.values()){
            if(value!=0){
                return false;
            }
        }
        return true;
    }
}